### DISCUSSION TITLE
How to use this type of Regular Expression in line filters

### DISCUSSION THREAD

**Discussion by fredtoussi** [WinMerge/winmerge#2253] [Q&A]
I am comparing Java source files and want to ignore the lines in which there are Java generic type declaration differences, for example:
    
     ArrayList<String> list = new ArrayList<>();    vs     ArrayList list = new ArrayList();

I am using <[A-Za-z]*> as the regular expression for the line filter, but it doesn't work. The same regular expression works fine in NotePad++ searches.

I have used other line filters which start with ^ (match from the start of line) and they work.

**Comment by sdottaka** [comment]
If both lines exist in a Diff hunk, the Line Filters pattern must match both lines to be applied.
In this example, you could use Substitution Filters instead, as shown below.

![image](https://github.com/WinMerge/winmerge/assets/98126/7ed09f1e-0946-4e09-af7f-b008b0dd8a6e)

**Comment by fredtoussi** [comment]
It works! 
Thank you very much for this useful app.

**Comment by fredtoussi** [comment]
I'm now having a problem with managing Substitution Filters. The original filter I created was working. I then modified the filter to cover multiple types but it didn't work. I removed the filters and created the original filter again. It didn't work. I have also tried various things such as restarting WinMerge in combination with removing and recreating the filter.

When a line filter is enabled/disabled etc. a message appears on the windows with the list of files: "Filters are updated. Do you want to refresh all open window compares. ...". This message does not appear when I enable/disable/add/remove a Substitution Filter.

I do not know where my filters are stored. The Edit > Options > System > Filter Folder setting was C:\Users\fs\Documents, but there is nothing related to WinMerge there. 

There is probably a bug. But if you can tell me how to find and remove the WinMerge Settings, so I can define and activate the original Substitution Filter, it will be of great help.

**Comment by sdottaka** [comment]
Substitution Filters are stored in the registry below.

`HKEY_CURRENT_USER\Software\Thingamahoochie\WinMerge\SubstitutionFilters`

**Reply by fredtoussi** [reply]
Thank you. I uninstalled and reinstalled WingMerge and succeeded in using the original filter. I'll edit the registry entry next time there is a problem.
